core/rest-client/src/http_client_builder.rs

```diff
@@ -0,0 +1,90 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::http_client::HttpClient;
+use http_req::response::Headers;
+use std::{string::String, time::Duration};
+
+/// Builder for `HttpClient`
+pub struct HttpClientBuilder {
+	/// Request timeout
+	timeout: Duration,
+
+	/// Send null body
+	send_null_body: bool,
+
+	/// pre-set headers
+	headers: Option<Headers>,
+
+	/// authorization
+	authorization: Option<String>,
+}
+
+impl Default for HttpClientBuilder {
+	fn default() -> Self {
+		Self {
+			timeout: Duration::from_secs(u64::MAX),
+			send_null_body: true,
+			headers: None,
+			authorization: None,
+		}
+	}
+}
+
+impl HttpClientBuilder {
+	/// Set request timeout
+	///
+	/// Default is no timeout
+	#[inline]
```

Comment:
I reckon this #[inline] is from the original implementation?

Even if yes, I still believe we should drop it. The best practice is to be very conservative about that and only use it, if it has heuristically been proven to improve performance in performance critical situations. The compiler is actually quite good at deciding itself when to inline code.

Reply:
Yes this is taken from `restson` - I'll remove it.